sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java

```diff
@@ -2684,6 +2685,52 @@ public void testWriteToTableDecorator() throws Exception {
     p.run();
   }
 
+  @Test
+  public void testBatchStorageWriteWithMultipleAppendsPerStream() throws Exception {
+    assumeTrue(useStorageApi);
```

Comment:
I do believe this can cause a test to silently never be executed if there is a misconfiguration of the parameterization. I could be wrong or out of date on this...

Reply:
The parameters are initialized here and all the tests in this class rely on them: https://github.com/apache/beam/blob/f903b9c6021a87af92c9bf03c01e1a5bf6a318d6/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java#L177-L186

Should we add another test that checks for existing configurations? e.g. check a configuration exists for streaming with storage api

Reply:
Hmm I guess we'd better just leave it, but just check that it executed for now. We might want to revisit the design of the test later.

Reply:
The test ran and passed for [Storage Write API](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Commit/2551/testReport/org.apache.beam.sdk.io.gcp.bigquery/BigQueryIOWriteTest/testBatchStorageWriteWithMultipleAppendsPerStream_2_/) and [Storage write at least once](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Commit/2551/testReport/org.apache.beam.sdk.io.gcp.bigquery/BigQueryIOWriteTest/testBatchStorageWriteWithMultipleAppendsPerStream_3_/)

Reply:
Btw not sure about Java, but many of our Python tests use the parameterized equivalent, (e.g. [here](https://github.com/apache/beam/blob/f0f4707bacf4f86dee344fb91978363f3eff7b12/sdks/python/apache_beam/io/gcp/bigquery_test.py#L1470-L1474), [here](https://github.com/apache/beam/blob/f0f4707bacf4f86dee344fb91978363f3eff7b12/sdks/python/apache_beam/io/filesystem_test.py#L125), [here](https://github.com/apache/beam/blob/f0f4707bacf4f86dee344fb91978363f3eff7b12/sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py#L742))

Reply:
Although yes, the way these tests are configured with `assumeTrue()`, many of them will be skipped silently (e.g. this test got skipped for [FILE_LOADS](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Commit/2551/testReport/org.apache.beam.sdk.io.gcp.bigquery/BigQueryIOWriteTest/testBatchStorageWriteWithMultipleAppendsPerStream_0_/)).